### ISSUE TITLE
perf_interp v2 site gate: Gemma-4 tp1 LM-head GEMM misses on every GPU database (silently dropped from Pareto search)

### ISSUE THREAD

**Issue by tianhaox** [ai-dynamo/aiconfigurator#1415]
## Summary

`perf_interp` v2's `ScatteredSites` gate (`max_site_distance = 2.0` octaves, joint log2 distance over `(n, k)`) makes **Gemma-4-26B-A4B's tp1 LM-head GEMM a structured miss on every GPU database in the repo** (66/66 system×backend×version combos checked). The failure is invisible today: the Pareto search swallows per-combination exceptions, so the support matrix still reports PASS while tp1 configurations are silently excluded from recommendations.

This is the same root cause as the B60/Llama-3.1-8B breakage fixed by the data patch in #1413, but it is not B60-specific and no data patch exists for it.

## Details

Gemma-4-26B-A4B has the largest vocab in `model_configs/`: `vocab_size = 262144`, `hidden_size = 2816`. The model queries its logits GEMM as `(n = vocab/tp, k = hidden)` (`sdk/models/gemma4.py:309,448`).

At tp1 the query site is `(262144, 2816)`:

- n axis: `log2(262144 / 65536) = 2.0` exactly (all GPU gemm tables cap n at 65536) — would pass on its own since the gate is `<=`.
- k axis: `hidden = 2816` is not a collected k grid point; nearest is 2560, adding `log2(2816/2560) = 0.1375` octaves.
- Joint distance: `sqrt(2.0^2 + 0.1375^2) = 2.0047 > 2.0` → `raise _miss(..., "no site within max_site_distance")` (`sdk/perf_interp/engine.py` scattered resolve; mirrored in Rust `perf_database/perf_interp.rs`).

Verified against every `systems/data/*/gemm/*/*/gemm_perf.parquet`: tp1 distance is ~2.004 on all GPU databases (n_max = 65536 everywhere); tp2 = 1.01, tp4 = 0.13 are fine. `b60/vllm/0.12.0` (still n_max = 12288) misses at tp1/tp2/tp4.

Sweep of all 68 model configs (tp1, H200 vLLM 0.24.0 sites):

| model | tp1 joint distance | status |
|---|---|---|
| google/gemma-4-26B-A4B (262144 / 2816) | **2.0039** | **MISS — all GPU databases** |
| Qwen3.5 family (248320) | 1.9218 | passes, brushing the gate |
| all other (vocab, hidden) combos | < 1.8 | fine |

Any future model with vocab > 262144 will miss at tp1 on every system.

## Why the support matrix doesn't catch it

`sdk/pareto_analysis.py` wraps each parallel config in `except Exception: ... "skip this combination"; continue`. The tp1 candidates raise `InterpolationDataNotAvailableError`, get dropped, and the search returns results from tp>=2 configs → support-matrix rows stay PASS. For a 26B A4B MoE, tp1 × N replicas is plausibly the actual tokens/s/gpu optimum, so recommendations are silently degraded rather than failing loudly. (The B60/Llama case in #1413 was visible only because at low GPU counts tp1/tp2 were the *only* candidates, so the result set came back empty.)

## Suggested fixes

1. **Mechanism (root fix):** make the frontier direction-aware in `ScatteredSites`. When the query site lies beyond the collected frontier in the scale-up direction (`n_q >= n_i`, `k_q >= k_i` for frontier sites), fall back to boundary util-hold (median util of the K log-nearest covering frontier sites, then `SOL(query)/util`) instead of a miss — the same mechanism already trusted unbounded along the m curve axis and for Grid ops' out-of-range. Keep the 2-octave gate for interior holes and the scale-down direction, where util genuinely doesn't transfer. Validate with a frontier-holdout LOO (hold out the largest-n sites), not the interior-hole LOO used to tune 2.0. Changes: Python `sdk/perf_interp/engine.py` + Rust `perf_database/perf_interp.rs` (both gate sites).
2. **Short-term data patch:** extend `output_feature_sizes` with vocab-derived grid points (adding n = 131072 alone brings gemma tp1 to ~1.0 octaves). Deriving the n axis from the supported-model vocab list (`vocab/tp` for tp in 1..8) prevents the next big-vocab model from re-tripping this.
3. **Guardrail:** a CI test that enumerates every supported model's LM-head `(vocab/tp, hidden)` shapes and asserts each system database resolves them (or fails the gate loudly), so coverage gaps surface at merge time instead of as silently-degraded recommendations.

## Repro

```python
import pandas as pd, glob, math
V, H = 262144, 2816  # google/gemma-4-26B-A4B
for p in sorted(glob.glob('aic-core/src/aiconfigurator_core/systems/data/*/gemm/*/*/gemm_perf.parquet')):
    df = pd.read_parquet(p, columns=['n', 'k'])
    sites = set(zip(df.n, df.k))
    d = min(math.sqrt(math.log2(V/sn)**2 + math.log2(H/sk)**2) for sn, sk in sites)
    print(f"{p.split('/data/')[1]:<40} tp1 dist={d:.4f} {'MISS' if d > 2.0 else 'ok'}")
```

Related: #1303 (perf_interp v2), #1413 (B60 LM-head data patch for Llama-3.1-8B).
